src/algo/HeapSort.js

```diff
@@ -508,15 +530,8 @@ export default class HeapSort extends Algorithm {
 
 	sortCallback() {
 		const list = this.listField.value.split(',').filter(x => x !== '');
-		if (
-			this.listField.value !== '' &&
-			list.length <= MAX_ARRAY_SIZE &&
-			list.map(Number).filter(x => x > 999 || Number.isNaN(x)).length <= 0
-		) {
-			this.implementAction(this.clear.bind(this));
-			this.listField.value = '';
-			this.implementAction(this.sort.bind(this), list);
-		}
+		this.implementAction(this.clear.bind(this));
```

Comment:
Just a nitpick, but I think it would be good to clear the infoLabel during "clear" like in other algorithms. If the user inputs a bad array then tries to run with a good array, the error message will still be present during the first 1 or 2 steps.

Reply:
Good catch, forgot to add that line of code.